### PR TITLE
Update version of Microsoft.Identity.Web and Microsoft.Identity.Web.UI to 2.7.0

### LINKED ISSUE #47823: Older versions of nuget packages for Blazor Webassembly hosted project
When creating a new Blazor Webassembly application with authentication.  The nuget package for Microsoft.Identity.Web and Microsoft.Identify.Web.UI  added with version 1.16.0 when the current version stands at 2.7.0. 

Any reasons for being that far behind? 

To replicate the issue: 

`dotnet new blazorwasm -au SingleOrg -ho --pwa -o testapp`

Check the testapp.Server.csproj  file, following references added.

`<PackageReference Include="Microsoft.Identity.Web" Version="1.16.0" />
    <PackageReference Include="Microsoft.Identity.Web.UI" Version="1.16.0" />`

### PR DESCRIPTION
**Pull request by surayya-MS**
# Update version of Microsoft.Identity.Web and Microsoft.Identity.Web.UI to 2.7.0

Fixes #47823

### CODE REVIEW

**Review by wtgodbe — APPROVED** [review]
If the transition across the major version is smooth, LGTM

**Comment by amcasey** [comment]
I see build failures because of `ObsoleteAttribute`, probably related to the major version bump.  Is there a minor version that meets our needs?

**Comment by surayya-MS** [comment]
> I see build failures because of `ObsoleteAttribute`, probably related to the major version bump. Is there a minor version that meets our needs?

@mkArtakMSFT , @halter73 should we update to the latest version (just checked it is 2.10.0) and deal with breaking changes or update to a minor version that doesn't create failures?

**Comment by surayya-MS** [comment]
Migration doc aka.ms/id-web-downstream-api-v2

File: src/ProjectTemplates/Web.ProjectTemplates/BlazorServerWeb-CSharp.csproj.in [hunk 1]
```diff
@@ -31,6 +31,7 @@
     <PackageReference Include="Microsoft.Identity.Web" Version="${MicrosoftIdentityWebVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
     <PackageReference Include="Microsoft.Identity.Web.MicrosoftGraph" Version="${MicrosoftIdentityWebMicrosoftGraphVersion}" Condition=" '$(GenerateGraph)' == 'True' " />
     <PackageReference Include="Microsoft.Identity.Web.UI" Version="${MicrosoftIdentityWebUIVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
+	<PackageReference Include="Microsoft.Identity.Web.DownstreamApi" Version="${MicrosoftIdentityWebDownstreamApiVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
```
Comment (halter73) [inline src/ProjectTemplates/Web.ProjectTemplates/BlazorServerWeb-CSharp.csproj.in:34]:
```suggestion
    <PackageReference Include="Microsoft.Identity.Web.DownstreamApi" Version="${MicrosoftIdentityWebDownstreamApiVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
```

File: src/ProjectTemplates/Web.ProjectTemplates/ComponentsWebAssembly-CSharp.Server.csproj.in [hunk 2]
```diff
@@ -45,6 +45,7 @@
     <PackageReference Include="Microsoft.Identity.Web" Version="${MicrosoftIdentityWebVersion}" />
     <PackageReference Include="Microsoft.Identity.Web.MicrosoftGraph" Version="${MicrosoftIdentityWebMicrosoftGraphVersion}" Condition=" '$(GenerateGraph)' == 'True' " />
     <PackageReference Include="Microsoft.Identity.Web.UI" Version="${MicrosoftIdentityWebUIVersion}" />
+	<PackageReference Include="Microsoft.Identity.Web.DownstreamApi" Version="${MicrosoftIdentityWebDownstreamApiVersion}" />
```
Comment (halter73) [inline src/ProjectTemplates/Web.ProjectTemplates/ComponentsWebAssembly-CSharp.Server.csproj.in:48]:
```suggestion
    <PackageReference Include="Microsoft.Identity.Web.DownstreamApi" Version="${MicrosoftIdentityWebDownstreamApiVersion}" />
```

File: src/ProjectTemplates/Web.ProjectTemplates/RazorPagesWeb-CSharp.csproj.in [hunk 3]
```diff
@@ -29,6 +29,7 @@
     <PackageReference Include="Microsoft.Identity.Web" Version="${MicrosoftIdentityWebVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
     <PackageReference Include="Microsoft.Identity.Web.MicrosoftGraph" Version="${MicrosoftIdentityWebMicrosoftGraphVersion}" Condition=" '$(GenerateGraph)' == 'True' " />
     <PackageReference Include="Microsoft.Identity.Web.UI" Version="${MicrosoftIdentityWebUIVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
+	<PackageReference Include="Microsoft.Identity.Web.DownstreamApi" Version="${MicrosoftIdentityWebDownstreamApiVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
```
Comment (halter73) [inline src/ProjectTemplates/Web.ProjectTemplates/RazorPagesWeb-CSharp.csproj.in:32]:
```suggestion
    <PackageReference Include="Microsoft.Identity.Web.DownstreamApi" Version="${MicrosoftIdentityWebDownstreamApiVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
```

File: src/ProjectTemplates/Web.ProjectTemplates/StarterWeb-CSharp.csproj.in [hunk 4]
```diff
@@ -29,6 +29,7 @@
     <PackageReference Include="Microsoft.Identity.Web" Version="${MicrosoftIdentityWebVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
     <PackageReference Include="Microsoft.Identity.Web.MicrosoftGraph" Version="${MicrosoftIdentityWebMicrosoftGraphVersion}" Condition=" '$(GenerateGraph)' == 'True' " />
     <PackageReference Include="Microsoft.Identity.Web.UI" Version="${MicrosoftIdentityWebUIVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
+	<PackageReference Include="Microsoft.Identity.Web.DownstreamApi" Version="${MicrosoftIdentityWebDownstreamApiVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
```
Comment (halter73) [inline src/ProjectTemplates/Web.ProjectTemplates/StarterWeb-CSharp.csproj.in:32]:
```suggestion
    <PackageReference Include="Microsoft.Identity.Web.DownstreamApi" Version="${MicrosoftIdentityWebDownstreamApiVersion}" Condition=" '$(IndividualB2CAuth)' == 'True' OR '$(OrganizationalAuth)' == 'True'" />
```

**Review by halter73 — COMMENTED** [review]
@dotnet/dnceng It looks like https://www.nuget.org/packages/Microsoft.Identity.Web.DownstreamApi is not in our `dotnet-public` feed. How do we go about adding it?

**Comment by surayya-MS** [comment]
> @dotnet/dnceng It looks like https://www.nuget.org/packages/Microsoft.Identity.Web.DownstreamApi is not in our `dotnet-public` feed. How do we go about adding it?

@wtgodbe , do you know  by any chance how to add a new package to our dotnet-public feed?

**Comment by wtgodbe** [comment]
>  It looks like https://www.nuget.org/packages/Microsoft.Identity.Web.DownstreamApi is not in our dotnet-public feed. How do we go about adding it?

https://github.com/dotnet/arcade/blob/main/Documentation/MirroringPackages.md

**Pull request merged**